subsys/bluetooth/audio/pacs.c

```diff
@@ -820,11 +996,16 @@ static int pacs_gatt_notify(struct bt_conn *conn,
 	params.func = pacs_gatt_notify_complete_cb;
 
 	/* Mark notification in progress */
-	atomic_set(&notify_rdy, 1);
+	if (atomic_test_bit(&pacs.flags, PACS_FLAG_UNREGISTERING) ||
+	    !atomic_test_bit(&pacs.flags, PACS_FLAG_REGISTERED)) {
+		return 0;
```

Comment:
Don't we need this check for all API functions to ensure that we don't attempt to modify values while the service is either unregistered or changing?

Reply:
We don't modify the values anywhere, except in `configure_pacs_char`, which is guarded by `PACS_FLAG_REGISTERED`, and only reset once `bt_pacs_unregister` is complete.

The only thing I can see that could happen at the same time, is if `bt_pacs_register` is called, and then `bt_pacs_unregister` is called while it executes, with a higher priority. This might need a guard...